hypothesis-python/src/hypothesis/internal/conjecture/junkdrawer.py

```diff
@@ -268,8 +268,13 @@ def __underlying_index(self, i: int) -> int:
                 i += 1
         return i
 
+    # even though we have len + getitem, mypyc requires iter.
+    def __iter__(self) -> Iterator[T]:
+        for i in range(len(self)):
+            yield self[i]
```

Comment:
I think this is technically an iterable, not an iterator?

Reply:
I think generators are possibly both? But I'm more confident they are an iterator since [they define `__iter__`](https://docs.python.org/3/library/stdtypes.html#generator-types). Here is cypthon typing generator-based `__iter__` both ways: https://grep.app/search?q=def%20__iter__%28self%29%20-%3E)

Also mypyc may be more strict on `__iter__` than cypthon:

> CPython implementation detail: CPython does not consistently apply the requirement that an iterator define [__iter__()](https://docs.python.org/3/library/stdtypes.html#iterator.__iter__). And also please note that the free-threading CPython does not guarantee the thread-safety of iterator operations.
> https://docs.python.org/3/glossary.html#term-iterator

so I think sticking with Iterator is best